library/base/src/main/java/org/cru/godtools/base/Settings.kt

```diff
@@ -74,6 +77,9 @@ class Settings @Inject internal constructor(
         prefs.getStringLiveData(PREF_PRIMARY_LANGUAGE, defaultLanguage.toLanguageTag()).distinctUntilChanged()
             .map { it?.toLocale() ?: defaultLanguage.also { primaryLanguage = it } }
     }
+    val primaryLanguageFlow
+        get() = prefs.getStringFlow(PREF_PRIMARY_LANGUAGE, defaultLanguage.toLanguageTag()).distinctUntilChanged()
+            .map { it?.toLocale() ?: defaultLanguage.also { primaryLanguage = it } }
```

Comment:
So is the ```distinctUntilChanged()``` what is listening for changes to the ```getStringFlow(PREF_PRIMARY_LANGUAGE, defaultLanguage.toLanguageTag())``` flow and then updating this flow accordingly? And this flow just returns that value that is set and ```val primaryLanguage``` is the value that is actually used?

Reply:
1. `getStringFlow()` will emit the current value of the shared preference as it is changed internally.
2. `distinctUntilChanged()` will then mutate the flow and only emit the value when it changed from the previous value.
3. then the `String` value that is being emitted is mutated again to convert it to a `Locale` object
4. If that `Locale` object is null, we then emit `defaultLanguage` instead, and in the process update the stored `primaryLanguage`

Reply:
in reality I should probably move the `distinctUntilChanged()` flow operator to be last, that way it will eliminate a duplicate value if the underlying `primaryLanguage` preference is ever set to null